Create github-upload-findings-in-security-dashboard.md (#1138)
returntocorp/semgrep-docs: docs/kb/semgrep-ci/github-upload-findings-in-security-dashboard.md

---
description: To prevent "resource not accessible by integration" error when running job to upload findings to GitHub's Advanced Security Dashboard
tags:
  - Semgrep CI
  - GitHub
---

## Why aren't findings populating in the GitHub Advanced Security Dashboard after running Semgrep in CI?

> When scanning with Semgrep in CI, findings automatically populate in the Semgrep Cloud Platform. To show findings in the GitHub Advanced Security Dashboard, you must first run an alternate job that uploads findings in the form of a `SARIF` file to the dashboard. See  [Sample GitHub Actions configuration file](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) for an example.

Reviewer: ```suggestion
When scanning with Semgrep in CI, findings automatically populate in Semgrep Cloud Platform. To show findings in the GitHub Advanced Security Dashboard, you must first run an alternate job that uploads findings in the form of a `SARIF` file to the dashboard. See  [Sample GitHub Actions configuration file](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) for an example.
```

Two threads about using the: 
* [Articles for our products](https://semgrepinc.slack.com/archives/C02JT6YA5S7/p1684444217271019)
* [A more in-depth look at "the"](https://semgrepinc.slack.com/archives/C02JT6YA5S7/p1684444774683329)

Once you run the alternate job, the job may fail and return a "resource not accessible by integration" error which has two possible causes:

> **1. Your repo is private**

Reviewer 2: ```suggestion
## Your repository is private
```
I also prefer repo, but our style guide generally prefers full usages at least once first - so you can do `repository (repo)` or just go with repository.

For the list > heading, this will make the typography more prominent and avoid suggesting that the issues are sequenced when they're actually independent.

Reviewer: <img width="738" alt="Snag_4807f1f" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/8065e910-d9db-4432-aa53-6fde052d93e3">

The reason for that convention has partly to do with writing for a global audience. Not all of our users speak English natively, so spelling out the whole word is easier for them if they use a translation service such as Google Translate or are unfamiliar with informal English.

Here's the wordlist source: [Google word list](https://developers.google.com/style/word-list)

> Third-party code scanning findings can only populate in the Advanced Security Dashboard if the repo is public/open source. Otherwise, the job will not have sufficient permissions to write to the dashboard regardless of any permissions set at the workflow or job level.

Reviewer 2: I would just delete open source here provided public is the relevant GH distinction.

> **2. The default workflow permissions in your repo are set to read only**

Reviewer 2: ```suggestion
## The workflow permissions in your repository are set to read-only
```

> Permissions typically default to more restrictive scopes, therefore, your workflow permissions will be set to read only unless changed manually.

Reviewer 2: ```suggestion
Your workflow permissions will be set to read-only (default) unless they've previously been changed.
```

> `org/repo -> Settings -> Actions -> General -> Workflow permissions`

Reviewer 2: @s-santillan How do we usually do this kind of selection hierarchy? I don't think we use code markup but I'm not sure what to suggest instead.

> `org/repo -> Settings -> Actions -> General -> Workflow permissions`

Reviewer: ```suggestion
To change permissions:
1. Navigate to your organization or repository in GitHub.
2. Click **Settings > Actions > General > Workflow permissions**.
```

Here's the convention: [Multi action procedures](https://developers.google.com/style/procedures#multi-action-procedures)

> ![image info](/img/kb/github-default-workflow-permissions.png)

Reviewer 2: I'd add a caption or other informational note here indicating these are the desired permissions.

> Bear in mind that changing the repo’s default workflow permissions changes the permissions for all workflows in your repo. So, for more granular permissions, utilize the `permissions` key either at the workflow or job level in the `semgrep.yml` workflow file. Learn more about the `permissions` key [here](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#setting-the-github_token-permissions-for-all-jobs-in-a-workflow), and check out our example of workflow level permissions below. Note that it only requires `security-events` write permissions to be successful.

Reviewer 2: ```suggestion
:::info 
Changing the repository's default workflow permissions changes the permissions for all workflows in that repository. For more granular permissions, set the `permissions` key at the workflow or job level in the `semgrep.yml` workflow file. Learn more about the `permissions` key at [Assigning permissions to jobs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#setting-the-github_token-permissions-for-all-jobs-in-a-workflow), or review the example workflow-level permissions below.
:::
```

Reviewer 2: `:::info ... :::` will create one of those nice little info boxes you see elsewhere in the docs.

> ## Example configuration of job to upload findings to Advanced Security Dashboard with permissions key

Reviewer 2: ```suggestion
### Example job configuration with `permissions` key

This job only requires `write` permissions for `security-events`.
```

```

> # Name of this GitHub Actions workflow.

Reviewer 2: I'd remove the comments in this snippet to keep it short - you can place a link back to the standard snippet for anyone who's less familiar.

---
description: To prevent "resource not accessible by integration" error when running job to upload findings to GitHub's Advanced Security Dashboard
tags:
  - Semgrep CI
  - GitHub
---

> # Uploading-findings-to-GitHub Advanced Security Dashboard

Reviewer: ```suggestion
# Uploading findings to GitHub Advanced Security Dashboard
```
I think the dashes were unintentional? :D

> 

Reviewer: Hey @pjzcodes ! We use the Google Style Guide (I see that you're already following a lot of the conventions!). For headings, I'd just like to add that we don't leave headings empty, that is, headings without at least an introductory sentence. 

Citation: [https://developers.google.com/style/headings#heading-and-title-format](https://developers.google.com/style/headings#heading-and-title-format)

Screenshot: 
<img width="550" alt="Snag_3cfaeb8" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/544868f8-258d-4f31-bbae-999427c1dbb4">

## Why aren't findings populating in the GitHub Advanced Security Dashboard after running Semgrep in CI?

> When scanning with Semgrep in CI, findings automatically populate in the Semgrep Cloud Platform. To show findings in the GitHub Advanced Security Dashboard, you must first run an alternate job that uploads findings in the form of a `SARIF` file to the dashboard. Click [here](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) for an example.

Reviewer: ```suggestion
When scanning with Semgrep in CI, findings automatically populate in the Semgrep Cloud Platform. To show findings in the GitHub Advanced Security Dashboard, you must first run an alternate job that uploads findings in the form of a `SARIF` file to the dashboard. See  [Sample GitHub Actions configuration file](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) for an example.
```

Ref: [Google style guide - writing link text](https://developers.google.com/style/link-text#write-link-text)
